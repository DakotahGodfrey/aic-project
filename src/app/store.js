import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
