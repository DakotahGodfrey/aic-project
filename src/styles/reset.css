*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
