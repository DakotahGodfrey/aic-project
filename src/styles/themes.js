const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
