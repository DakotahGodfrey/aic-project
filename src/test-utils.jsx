import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
