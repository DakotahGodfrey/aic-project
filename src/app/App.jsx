import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
