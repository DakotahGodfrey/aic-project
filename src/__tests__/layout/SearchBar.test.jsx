import { store } from 'app/store';
import SearchBar from 'components/layout/Search';
import { defaultTheme } from 'styles/themes';
import { render, screen, fireEvent } from 'test-utils';

const placeholder = 'Search';
const searchHandler = jest.fn();
let value = '';
const changeHandler = jest.fn();

describe('<SearchBar />', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders an input with type search and the correct placeholder', () => {
    render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    expect(screen.getByRole('searchbox')).toHaveAttribute(
      'placeholder',
      placeholder
    );
  });
  it('renders an input with type search and the correct value', () => {
    render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    expect(screen.getByRole('searchbox')).toHaveAttribute('value', value);
  });
  it('calls the appropriate handler when the input changes', () => {
    render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    fireEvent.change(screen.getByRole('searchbox'), { target: { value: 'a' } });
    expect(changeHandler).toHaveBeenCalledTimes(1);
  });
  it('renders a submit button that is disabled when the input is empty', () => {
    const { rerender } = render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    expect(screen.getByRole('button', { name: 'search' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'search' })).toHaveAttribute(
      'disabled'
    );
  });
  it('calls the appropriate handler when the submit button is clicked', () => {
    let value = 'sdfsd';
    render(
      <SearchBar
        placeholder={placeholder}
        searchHandler={searchHandler}
        value={value}
        changeHandler={changeHandler}
      />,
      {
        theme: defaultTheme,
        store: store,
      }
    );
    fireEvent.click(screen.getByRole('button', { name: 'search' }));
    expect(searchHandler).toHaveBeenCalled();
  });
});
