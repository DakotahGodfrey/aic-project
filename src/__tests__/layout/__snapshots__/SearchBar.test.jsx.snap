// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> matches the snapshot 1`] = `
<form
  class="sc-bdnxRM hGSUJh"
>
  <input
    placeholder="Search"
    required=""
    type="search"
    value=""
  />
  <button
    class="material-icons"
    disabled=""
    type="submit"
  >
    search
  </button>
</form>
`;
